include/revng/TupleTree/TupleTree.h

```diff
@@ -25,11 +25,30 @@
 #include "revng/Support/Assert.h"
 #include "revng/Support/Debug.h"
 #include "revng/Support/YAMLTraits.h"
+#include "revng/TupleTree/Tracking.h"
 #include "revng/TupleTree/TupleTreeCompatible.h"
 #include "revng/TupleTree/TupleTreePath.h"
 #include "revng/TupleTree/TupleTreeReference.h"
 #include "revng/TupleTree/Visits.h"
 
+template<typename T>
+concept HasTracking = requires(T _) { T::HasTracking; };
+
+template<typename T>
+struct TrackGuard {
+  const T *TrackedObject;
+
+public:
+  TrackGuard(const T &TrackedObject) : TrackedObject(&TrackedObject) {
+    if constexpr (HasTracking<T>)
```

Comment:
Why would we want to `TrackGuard` something that doesn't have tracking?

Reply:
because this mechanism ties to the optionality of emitting tracking information from jinja generated classes. i did not wanted to leak out from here this feature of the generation, so i hide it in here.

Reply:
I'm not sure about the impact of this choice, but at a minimum add a comment explaining this.
It's very surprising.